project_generator/project.py

```diff
@@ -427,71 +403,40 @@ def generate_dict_for_tool(self, tool):
+        self.project['linker_file'] =  self.project['linker_file'] or toolchain_specific_settings.linker_file or [
+            tool_settings.linker_file for tool_settings in tool_specific_settings if tool_settings.linker_file]
 
-            'linker_file': self.linker_file or toolchain_specific_settings.linker_file or [
-                tool_settings.linker_file for tool_settings in tool_specific_settings if tool_settings.linker_file],
+        self.project['macros'] = self.project['macros'] + list(flatten([settings.macros for settings in tool_specific_settings])) \
+            + toolchain_specific_settings.macros
 
-            'macros': self.macros + list(flatten([
-                settings.macros for settings in tool_specific_settings])) + toolchain_specific_settings.macros,
+        self.project['misc'] =  [merge_recursive({k: v for settings in tool_specific_settings for k, v in settings.misc.items()},
+            toolchain_specific_settings.misc)]
 
-            'misc': [
-                merge_recursive({k: v for settings in tool_specific_settings for k, v in settings.misc.items()},
-                                toolchain_specific_settings.misc)],
-            'project_dir': self.project_dir,
+        self.project['template'] = toolchain_specific_settings.template or [
+                tool_settings.template for tool_settings in tool_specific_settings if tool_settings.template]
 
-            'template': toolchain_specific_settings.template or [
-                tool_settings.template for tool_settings in tool_specific_settings if tool_settings.template],
-        }
-        if d['linker_file'] == None and d['output_type'] == 'exe':
+        if self.project['linker_file'] == None and self.project['output_type'] == 'exe':
             raise RuntimeError("Executable - no linker command found.")
+
         if self.workspace.settings.generated_projects_dir != self.workspace.settings.generated_projects_dir_default:
-            output_dir = self.workspace.settings.generated_projects_dir
-            output_dir = output_dir.replace('$tool$', tool)
-            output_dir = output_dir.replace('$project_name$', self.name)
-            if self.target:
-                output_dir = output_dir.replace('$target$', self.target)
-            workspace = self._get_workspace_name()
-            if workspace:
-                output_dir = output_dir.replace('$workspace$', workspace)
+            output_dir = Template(self.workspace.settings.generated_projects_dir)
+            output_dir = output_dir.substitute(target=self.project['target'], workspace=self._get_workspace_name(),
+                                               project_name=self.name, tool=tool)
         else:
-            output_dir = os.path.join(self.project_dir['path'], "%s_%s" % (tool, self.name))
-        d['output_dir']['path'] = os.path.normpath(output_dir)
-        return d
-
-    def fixup_executable(executable_path, tool):
```

Comment:
why this one is removed? I'll show you valinor today which uses this, we need to find out why it was added and do clean-up ;)